### PR TITLE
Add shell-based application 

### PR DESCRIPTION
**Pull request by timClicks**
Uses `writeShellApplication` rather than `mkDerivation` to create a shell-only application.

### CODE REVIEW

File: hello-shell/default.nix [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# This example uses a different pattern than many of the
+# others in this repository, by relying on a "trivial builder"
+#
+# See the manual for more details:
+#
+# <https://nixos.org/manual/nixpkgs/stable/#trivial-builder-writeShellApplication>
+
+{writeShellApplication}:
```
Comment (lovesegfault) [inline hello-shell/default.nix]: style: We leave spaces around braces
```suggestion
{ writeShellApplication }:
```

You can use `nixpkgs-fmt` to do this for you:
```console
$ nix run nixpkgs#nixpkgs-fmt -- .
```

File: hello-shell/default.nix [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+writeShellApplication {
+    name = "hello-shell";
+
+    runtimeInputs = [];
```
Comment (lovesegfault) [inline hello-shell/default.nix]: style: Spacing
```suggestion
    runtimeInputs = [ ];
```

File: hello-shell/default.nix [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+    text = ''
+      echo Hello, shell world!
```
Comment (lovesegfault) [inline hello-shell/default.nix]: Does the `!` work as expected without quoting?

**Review by lovesegfault — COMMENTED** [review]
Awesome!

File: hello-shell/default.nix (hunk 3 above)
Comment (timClicks) [reply hello-shell/default.nix]: Seems to?

```console
$ pwd
/home/tim/Open/nix-workshop
$ nix run
warning: Git tree '/home/tim/Open/nix-workshop' is dirty
Hello, C world!
Hello, Cobol world!
Hello, Go World!
Hello, Poetry world!
Hello, Rust world!
Hello, Setuptools world!
Hello, shell world!
```

File: hello-shell/default.nix (hunk 1 above)
Comment (timClicks) [reply hello-shell/default.nix]: Boo that's lame. Fixed.

**Comment by timClicks** [comment]
I've updated the README within the merge commit.

**Comment by lovesegfault** [comment]
Thanks Tim! Great attention to detail keeping the docker image hash up-to-date in the docs :)

**Pull request merged**